app/models/container.rb

```diff
@@ -27,16 +27,17 @@ class Container < ApplicationRecord
 
   acts_as_miq_taggable
 
-  def event_where_clause(assoc = :ems_events)
-    case assoc.to_sym
-    when :ems_events, :event_streams
-      # TODO: improve relationship using the id
-      ["container_namespace = ? AND #{events_table_name(assoc)}.ems_id = ? AND container_name = ?",
-       container_project.try(:name), ext_management_system.try(:id), name]
-    when :policy_events
-      # TODO: implement policy events and its relationship
-      ["#{events_table_name(assoc)}.ems_id = ?", ext_management_system.try(:id)]
-    end
+  def event_where_clause_ems_events
+    # TODO: improve relationship using the id
+    EmsEvent
+      .where(:container_name => name)
+      .where(:container_namespace => container_project.try(:name))
+      .where(:ems_id => ext_management_system.try(:id))
```

Comment:
```suggestion
      .where(:ems_id => ext_management_system&.id)
```

not sure if this does what we want.
If ems is nil, then it will look for all records with a nil ems_id rather than not looking at ems_id.

If that is what you want then cool.

nvr mind: looks like that is what is in there from before. Again, not sure it is what we want but is better to be bug for bug compatible until we know we want otherwise

Reply:
Yeah, I have been tempted to kill things that are obviously wrong.  I'll create a test based on your example and see if I can find a better solution.